lib/sys-info.ts

```diff
@@ -311,26 +291,43 @@ export class SysInfo implements NativeScriptDoctor.ISysInfo {
 
 			if (shouldUseXcproj) {
 				// If that's the case we can use xcproj gem to convert them back to ASCII plist format
-				xcprojAvailable = !!(await this.exec("xcproj --version"))
+				xcprojAvailable = !!(await this.exec("xcproj --version"));
 			}
 
-			this.xcprojInfoCache = { shouldUseXcproj, xcprojAvailable };
-		}
-
-		return this.xcprojInfoCache;
+			return { shouldUseXcproj, xcprojAvailable };
+		});
 	}
 
-	public async isCocoaPodsUpdateRequired(): Promise<boolean> {
-		if (this.isCocoaPodsUpdateRequiredCache === null) {
+	public isCocoaPodsUpdateRequired(): Promise<boolean> {
+		return this.getValueForProperty(() => this.isCocoaPodsUpdateRequiredCache, async (): Promise<boolean> => {
 			let xcprojInfo = await this.getXcprojInfo();
 			if (xcprojInfo.shouldUseXcproj && !xcprojInfo.xcprojAvailable) {
-				this.isCocoaPodsUpdateRequiredCache = true;
+				return true;
 			} else {
-				this.isCocoaPodsUpdateRequiredCache = false;
+				return false;
 			}
-		}
+		});
+	}
+
+	public setShouldCacheSysInfo(shouldCache: boolean): void {
+		this.shouldCache = shouldCache;
+	}
+
+	private async getValueForProperty<T>(property: Function, getValueMethod: () => Promise<T>): Promise<T> {
+		if (this.shouldCache) {
+			const propertyName = this.helpers.getPropertyName(property);
+			const cachedValue: T = (<any>this)[propertyName];
 
-		return this.isCocoaPodsUpdateRequiredCache;
+			if (cachedValue === undefined || cachedValue === null) {
```

Comment:
most of the methods that should be cached, return `null`, for example `getJavaCompilerVersion`. Does it mean that if I have a problem with Javac, each execution will not  be cached?